odoo/session.go

```diff
@@ -0,0 +1,70 @@
+package odoo
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+var (
+	// ErrInvalidCredentials is an error that indicates an authentication error due to missing or invalid credentials.
+	ErrInvalidCredentials = errors.New("invalid credentials")
+)
+
+// Session information
+type Session struct {
+	// ID is the session ID.
+	// Is always set, no matter the authentication outcome.
+	ID string `json:"session_id,omitempty"`
+	// UID is the user's ID as an int, or the boolean `false` if authentication failed.
+	UID int `json:"uid,omitempty"`
+}
+
+type loginParams struct {
+	DB       string `json:"db,omitempty"`
+	Login    string `json:"login,omitempty"`
+	Password string `json:"password,omitempty"`
+}
+
+// Login tries to authenticate the user against Odoo.
+// It returns a session if authentication was successful. An error is returned if
+//  - the credentials were wrong,
+//  - encoding or sending the request,
+//  - or decoding the request failed.
+func (c Client) Login(login, password string) (*Session, error) {
+	// Prepare request
+	body, err := NewJSONRPCRequest(loginParams{c.db, login, password}).Encode()
+	if err != nil {
+		return nil, fmt.Errorf("encoding request: %w", err)
+	}
+
+	// Send request
+	res, err := c.http.Post(c.baseURL+"/web/session/authenticate", "application/json", body)
+	if err != nil {
+		return nil, fmt.Errorf("login: sending HTTP request: %w", err)
+	}
```

Comment:
I would prefer to pass a context here. This would allow us to easily cancel long requests through the context.
```suggestion
    req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.baseURL+"/web/session/authenticate", body)
	if err != nil {
		return nil, fmt.Errorf("login: building HTTP request: %w", err)
	}
	// Send request
	res, err := c.http.Do(req)
	if err != nil {
		return nil, fmt.Errorf("login: sending HTTP request: %w", err)
	}
```